### PR TITLE
TELCODOCS-1825: Add a webhook for namespace deletion

### PR DESCRIPTION
**Pull request by StephenJamesSmith**
D/S Docs: [MGMT-17316] Add a webhook for namespace deletion

Jira: https://issues.redhat.com/browse/TELCODOCS-1825

Version(s): openshift-4.16.0, openshift-4.15.0, openshift-4.14.0

Link to docs preview: https://75412--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hardware_enablement/kmm-kernel-module-management.html#kmm-unloading-kernel-module_kernel-module-management-operator

SME review: @qbarrand
QE review: @cdvultur

### CODE REVIEW

File: modules/kmm-unloading-kernel-module.adoc [hunk 1]
```diff
@@ -16,8 +16,5 @@ You must unload the kernel modules when moving to a newer version or if they int
 ====
 When unloading worker pods, KMM needs all the resources it uses when loading the kernel module. This includes the `ServiceAccount` referenced in the `Module` as well as any RBAC defined to allow privileged KMM worker Pods to run. It also includes any pull secret referenced in `.spec.imageRepoSecret`.
 
-To avoid situations where KMM is unable to unload the kernel module from nodes:
-
-* Do not delete those resources while the `Module` resource is still present in the cluster in any state, including `Terminating`.
-* Do not delete any namespace containing at least a `Module` resource.
+To avoid situations where KMM is unable to unload the kernel module from nodes, make sure those resources are not deleted while the `Module` resource is still present in the cluster in any state, including `Terminating`. KMM ships with a validating admission webhook that rejects the deletion of namespaces that contain at least one `Module` resource.
```
Comment (StephenJamesSmith) [inline modules/kmm-unloading-kernel-module.adoc:19]: @cdvultur Please review the above - basically it rewords the existing text and adds mention of webhook. Also, if I've interpreted https://github.com/rh-ecosystem-edge/kernel-module-management/commit/6e1902da520f894b82bc41257ec15242a6973ef1 correctly, this is the only section ([docs/mkdocs/documentation/deploy_kmod.md]) intended for cusstomer-facing documentation. Please confirm.

Comment (cdvultur) [reply modules/kmm-unloading-kernel-module.adoc:19]: /lgtm

[the remark this answers is not in the document]
Comment (StephenJamesSmith) [reply modules/kmm-unloading-kernel-module.adoc:19]: changed.

**Review by jldohmann — APPROVED** [review]
LGTM!

**Comment by StephenJamesSmith** [comment]
@jldohmann Not sure I understand - Havee all cherrypicks been done and are succesful pr is there more to be done?

**Pull request merged**